packages/@aws-cdk/aws-apigateway/lib/api-key.ts

```diff
@@ -20,6 +20,7 @@ export interface IApiKey extends IResourceBase {
  */
 export interface ApiKeyProps extends ResourceOptions {
   /**
+   * [disable-awslint:ref-via-interface]
```

Comment:
Why is this needed? We generally don't encourage any new linter exclusions.

Reply:
The build was failing after I made RateLimitedApiKeyProps extend ApiKeyProps :

```
error: [awslint:ref-via-interface:@aws-cdk/aws-apigateway.RateLimitedApiKeyProps.resources] API should use interface and not the concrete class (@aws-cdk/aws-apigateway.RestApi). If this is intentional, add "[disable-awslint:ref-via-interface]" to element's jsdoc
```
So I followed the suggestion and added ```[disable-awslint:ref-via-interface]``` to the jsdoc of ```resources``` property in ```ApiKeyProps```

Reply:
I see it now.

I believe this is occurring because this object is using `RestApi` instead of `IRestApi` as the type for `resources`. 
Can we try switching it to `IRestApi` and see if all tests still pass?

Reply:
I tried doing that, but it resulted in compilation errors as ```renderStageKeys``` method expects ```resources``` to be of type ```RestApi``` and changing it to ```IRestApi``` is not possible as it doesn't have ```deploymentStage``` property needed by this method 
https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-apigateway/lib/api-key.ts#L90-L101